agent/exploits/jetpack_version_detection.py

```diff
@@ -0,0 +1,202 @@
+    ("9.9.0", "9.9.2"),
+    ("9.8.0", "9.8.2"),
+    ("9.7.0", "9.7.2"),
+    ("9.6.0", "9.6.3"),
+    ("9.5.0", "9.5.4"),
+    ("9.4.0", "9.4.3"),
+    ("9.3.0", "9.3.4"),
+    ("9.2.0", "9.2.3"),
+    ("9.1.0", "9.1.2"),
+    ("9.0.0", "9.0.4"),
+    ("8.9.0", "8.9.3"),
+    ("8.8.0", "8.8.4"),
+    ("8.7.0", "8.7.3"),
+    ("8.6.0", "8.6.3"),
+    ("8.5.0", "8.5.2"),
+    ("8.4.0", "8.4.4"),
+    ("8.3.0", "8.3.2"),
+    ("8.2.0", "8.2.5"),
+    ("8.1.0", "8.1.3"),
+    ("8.0.0", "8.0.2"),
+    ("7.9.0", "7.9.3"),
+    ("7.8.0", "7.8.3"),
+    ("7.7.0", "7.7.5"),
+    ("7.6.0", "7.6.3"),
+    ("7.5.0", "7.5.6"),
+    ("7.4.0", "7.4.4"),
+    ("7.3.0", "7.3.4"),
+    ("7.2.0", "7.2.4"),
+    ("7.1.0", "7.1.4"),
+    ("7.0.0", "7.0.4"),
+    ("6.9.0", "6.9.3"),
+    ("6.8.0", "6.8.4"),
+    ("6.7.0", "6.7.3"),
+    ("6.6.0", "6.6.4"),
+    ("6.5.0", "6.5.3"),
+    ("6.4.0", "6.4.5"),
+    ("6.3.0", "6.3.6"),
+    ("6.2.0", "6.2.4"),
+    ("6.1.0", "6.1.4"),
+    ("6.0.0", "6.0.3"),
+    ("5.9.0", "5.9.3"),
+    ("5.8.0", "5.8.3"),
+    ("5.7.0", "5.7.4"),
+    ("5.6.0", "5.6.4"),
+    ("5.5.0", "5.5.4"),
+    ("5.4.0", "5.4.3"),
+    ("5.3.0", "5.3.3"),
+    ("5.2.0", "5.2.4"),
+    ("5.1.0", "5.1.3"),
+    ("5.0.0", "5.0.2"),
+    ("4.9.0", "4.9.2"),
+    ("4.8.0", "4.8.4"),
+    ("4.7.0", "4.7.3"),
+    ("4.6.0", "4.6.2"),
+    ("4.5.0", "4.5.2"),
+    ("4.4.0", "4.4.4"),
+    ("4.3.0", "4.3.4"),
+    ("4.2.0", "4.2.4"),
+    ("4.1.0", "4.1.3"),
+    ("4.0.0", "4.0.6"),
+    (None, "3.9.9"),
+]
+
+
+@exploits_registry.register
+class JetpackExploit(webexploit.WebExploit):
+    accept_request = definitions.Request(
+        method="GET", path="/wp-content/plugins/jetpack/readme.txt"
+    )
+    check_request = definitions.Request(
+        method="GET", path="/wp-content/plugins/jetpack/readme.txt"
+    )
+    accept_pattern = [
+        re.compile("=== Jetpack - WP Security, Backup, Speed, & Growth ===")
+    ]
+    version_pattern = re.compile("Stable\stag:\s(\d+\.\d+\.\d+)")
+
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        """Rule to detect specific vulnerability on a specific target.
+
+        Args:
+            target: Target to scan
+
+        Returns:
+            List of identified vulnerabilities.
+        """
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        target_endpoint = urlparse.urljoin(target.origin, self.check_request.path)
+
+        try:
+            req = requests.Request(
+                method=self.check_request.method,
+                url=target_endpoint,
+                data=self.check_request.data,
```

Comment:
Do we need to raise for status? or is that not a case that can happen here?

Reply:
no